### ISSUE TITLE
Tests incompatible with Numpy 1.24

### ISSUE THREAD

**Issue by berland** [equinor/xtgeoapp-grd3dmaps#71]
With numpy bumped from 1.23 to 1.24, some numbers change making the tests fail:

Example failure:
```
    def test_aggregated_map2(datatree):
        result = datatree / "aggregate2_folder"
        result.mkdir(parents=True)
        cfg = "tests/yaml/aggregate2.yml"
    
        grid3d_aggregate_map.main(
            [
                "--config",
                cfg,
                "--mapfolder",
                str(result),
                "--plotfolder",
                str(result),
            ]
        )
        swat = xtgeo.surface_from_file(result / "all--min_swat--20030101.gri")
>       assert swat.values.mean() == pytest.approx(0.7908786104444353, abs=1e-8)
E       assert 0.7309275812612058 == 0.7908786104444353 ± 1.0e-08
E         comparison failed
E         Obtained: 0.7309275812612058
E         Expected: 0.7908786104444353 ± 1.0e-08
```

**Comment by vegardkv** [comment]
I've had a brief look at this, but haven't gotten to the root of the issues yet. I've tracked execution up until line 113 in grid3d_aggregate_map.py, and the results seem to be the same up to that point. This might therefore be caused by something in `_ndarray_to_regsurfs` or `_write_surfaces` in the same file, which again could be related to the mentioned compatibility issue in 1.24. I probably won't be able to dig any further until after new years, though.

**Comment by vegardkv** [comment]
After some more digging, this seems to be a bug that was introduced in `numpy` 1.24.0. I have filed a bug report: https://github.com/numpy/numpy/issues/22912. 

The error was not related to the above mentioned issue, but rather this one: https://numpy.org/devdocs/release/1.24.0-notes.html#masked-invalid-now-modifies-the-mask-in-place. `masked_invalid` is used by `xtgeo.RegularSurface.get_values1d`, which in this context is used when writing the surface to file (`_export_irap_binary_cxtgeo`). This should therefore have implications also for xtgeo.

**Comment by mferrera** [comment]
Resolved with #70 and then #78 (and in numpy)
